src/Workspaces/Core/Portable/Options/EditorConfig/EditorConfigStorageLocation.cs

```diff
@@ -30,9 +33,25 @@ private static Optional<bool> ParseBool(string str)
         private static readonly Func<string, Optional<int>> s_parseInt32 = ParseInt32;
         private static Optional<int> ParseInt32(string str)
             => int.TryParse(str, out var result) ? result : new Optional<int>();
+
+        private static readonly Func<string, Optional<string>> s_parseString = ParseString;
+        private static Optional<string> ParseString(string str)
+        {
+            if (str.Equals("unset", StringComparison.Ordinal))
+            {
+                return default;
+            }
+
+            str ??= "";
+            return str.Replace("\\r", "\r").Replace("\\n", "\n");
+        }
+
         private static readonly Func<int, string> s_getInt32EditorConfigStringForValue = GetInt32EditorConfigStringForValue;
         private static string GetInt32EditorConfigStringForValue(int value) => value.ToString().ToLowerInvariant();
 
+        private static readonly Func<string, string> s_getStringEditorConfigStringForValue = GetStringEditorConfigStringForValue;
+        private static string GetStringEditorConfigStringForValue(string value) => value.ToString().Replace("\r", "\\r").Replace("\n", "\\n");
```

Comment:
does this need to roundtrip `default` to `"unset"`?

Reply:
We don't need to round-trip. This direction is used for generating **.editorconfig** when one did not exist.

Reply:
I ended up needing to generate `unset` for the default value since the **.editorconfig** code generation expects the generated value to never be empty.

https://github.com/dotnet/roslyn/blob/5d247289d30d183ecbfaa7f59163ea33b915a36d/src/Workspaces/Core/Portable/Options/EditorConfig/EditorConfigStorageLocation%601.cs#L97